docs/decisions/0002-auto-enrollment-post-request-approval.rst

```diff
@@ -0,0 +1,158 @@
+not the course it's was redeemed for. We have to query ecommerce data to check the order created using the coupon code
+for that information. Automatically enrolling a learner after a request approval would guarantee that a code is used to enroll in
+the intended course.
+
+For license requests, it's less of a problem because a license can be used for all courses in an enterprise catalog and
+auto-enrollment would just save learners clicks for this case.
+
+Questions to ponder (this is a draft so I'm putting it here for visibility for now):
+* Does enrolling a learner after a request approval provide a good/better user experience?
+  The current flow for request approval gives learners the flexibility to choose when to enroll in a course.
+  A user might request access to multiple courses with the intention of taking them one at a time or sometime later.
+  Automatically enrolling them in multiple courses might force learners to bookmark courses or even unenroll.
+* How valuable is the guarantee that a code is used to enroll a learner in the intended course?
+  A learner can use a code intended for one course to enroll in another, but it's an edge case since the email we send
+  after a request approval links them to the request course.
+* Which course run do we enroll users in if there are multiple? Do we allow learners to choose or do we enroll them in the first available course run?
+* How does this impact analytics/downstream reporting? We need to provide a way for analytics to separate auto enrollments from regular enrollments.
+
+Decision
+============
+
+We're attempting to adopt event-based communication between the enterprise-access service, the LMS (possibly), and the services
+responsible for granting/redeeming a subsidy.
+
+The following events will be sent to an event bus when the state of a subsidy request is approved:
+* `org.openedx.enterprise.access.coupon-code-request.approved.v1`
+* `org.openedx.enterprise.access.license-request.approved.v1``
+These events will be produced via Django signals that live in the `openedx-events <https://github.com/openedx/openedx-events>`__ repo.
+
+Auto-enrollment will be a feature that could be toggled off as it does not always make sense to auto-enroll a learner,
+e.g. when a coupon code offers 50% discount, a license request without course id, etc.. We will add a field `attempt_auto_enrollment`
+to the `SubsidyRequestCustomerConfiguration` model to allow for the auto-enrollment to be turned on/off.
+We could also explore having the auto-enrollment be a request specific rather than an enterprise specific toggle.
+
+The request approval process will involve these steps:
+1. An admin approves a subsidy request.
+2. The state on the subsidy request changes to `approved`.
+3. An subsidy request approved event is sent to the event bus.
+4. license-manager or ecommerce will consume this event, assigns a subsidy, and send a subsidy assigned event.
```

Comment:
Would it be better to use fulfilled here instead of subsidy assignment? Help us align the language better.